src/Ormolu/Utils/Extensions.hs

```diff
@@ -21,6 +22,9 @@ import qualified Distribution.ModuleName as ModuleName
 import Distribution.PackageDescription
 import Distribution.PackageDescription.Parsec
 import qualified Distribution.Types.CondTree as CT
+#if MIN_VERSION_Cabal(3,6,0)
```

Comment:
Rather than introducing CPP, let's use `Cabal >=3.6 && <3.7`.

Reply:
Then Ormolu will then no longer be in Stackage, which was the reason why we did not use Cabal 3.6 earlier: https://github.com/tweag/ormolu/pull/759#discussion_r687553174 Personally, I don't care about Ormolu being in Stackage, especially as we have prebuild binaries.

Reply:
I see. Sorry, I forgot about that aspect. In that case let's keep CPP, but let's also move it down so it is sort of standalone and not in the middle of normal imports, so that normal imports can be sorted properly.

Reply:
Actually it seems that Stackage ties the version of `ghc-lib-parser` to the version of GHC. This means that they are going to update to `ghc-lib-paser-9.2.x.x` when they switch to GHC 9.2. This will force them also to switch to Cabal-3.6 if I understand correctly. This means that if we are switching to `ghc-lib-parser-9.2.x.x` we can as well simply switch to `Cabal >=3.6 && <3.7` too.

Reply:
I see, will change :+1: